### PR TITLE
Implement Solver class with factorial, reverse, and fizzbuzz methods using TDD

### PR DESCRIPTION
**Pull request by didierganthier**
## Summary

This PR implements the `Solver` class with three methods: `factorial`, `reverse`, and `fizzbuzz`. The `factorial` method calculates the factorial of a given positive integer, and raises an exception for negative integers. The `reverse` method returns the reversal of a given string. The `fizzbuzz` method returns "fizz" for numbers divisible by 3, "buzz" for numbers divisible by 5, "fizzbuzz" for numbers divisible by both 3 and 5, and the number itself as a string for all other cases.

## Checklist
- [x] I have written tests for the new code (using `rspec`)
- [x] I have added the tests to the appropriate test suite
- [x] I have written testable code that passes all tests
- [x] I have added relevant documentation to the code

## Changes Made
- Created `Solver` class with three methods
- Added `rspec` tests for the three methods
- Added relevant documentation to the code

## Run the tests
```
rspec spec/solver_spec.rb
```

## Authors:
- @didierganthier 
- @cvalencia1991

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -237,47 +189,25 @@ Example:
...
-
-- **[Question_2]**
-
-  - [Answer_2]
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-<!-- LICENSE -->
 
 ## 📝 License <a name="license"></a>
```
Comment (1GORDON) [inline README.md:212]: Great job team 🤝

- [ ] [Optional] Kindly fix the broken MIT license link to make your readme file more professional.

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @didierganthier & @cvalencia1991 ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (didierganthier) [reply README.md:212]: Thank you @1GORDON 👍🏽

**Review by addod19 — COMMENTED** [review]
### Hi @didierganthier, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because the link to the project is already approved and merged :+1: 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

You have submitted a project, that is invalid because the link to the project is already approved and merged 👍

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Pull request merged**